src/java.desktop/share/classes/com/sun/beans/introspect/PropertyInfo.java

```diff
@@ -91,6 +91,9 @@ private boolean initialize() {
                 if (writeType == null) {
                     this.write = info;
                     writeType = info.type;
+                } else if (isParentOfIncoming(this.write, info)) {
```

Comment:
Is it possible to update the logic for the read methods above and remove the sorting based on the default property?